src/utils/TedexSetup.ts

```diff
@@ -0,0 +1,59 @@
+import { TraderClient, TradeType } from 'tdex-sdk';
+import { toSatoshi } from './format';
+import BigNumber from 'bignumber.js';
+import { Market } from './constants';
+
+const client = new TraderClient('http://localhost:9945');
+
+export const handleMarkets = async () => {
+  try {
+    const markets = await client.markets();
+    console.log(markets);
+
+    return markets;
+  } catch (error) {
+    throw new Error('TDEX provider and markets is not reachable');
+  }
+};
+
+export const onSendAmountChange = async (
+  amount: number,
+  trade: TradeType,
+  asset: string,
+) => {
+  try {
+    const pricesPreview = await client.marketPrice(
+      {
+        baseAsset: Market.L_BTC,
+        quoteAsset: Market.L_USDT,
+      },
+      trade,
+      toSatoshi(amount),
+      asset,
+    );
+
+    console.log(pricesPreview);
+    
+
+    const price = pricesPreview[0].amount;
+
+    return new BigNumber(price);
```

Comment:
this should be converted back to fractional number (ie. fromSatoshi) using the opposite asset precision as well.

Reply:
fixed this